provides.py

```diff
@@ -0,0 +1,23 @@
+from charmhelpers.core import hookenv
+from charms.reactive import when, when_not
+from charms.reactive import set_flag, clear_flag
+from charms.reactive import Endpoint
+
+
+class MountProvides(Endpoint):
+
+    @when('endpoint.{endpoint_name}.joined')
+    def joined(self):
+        set_flag(self.expand_name('{endpoint_name}.available'))
+
+    @when_not('endpoint.{endpoint_name}.joined')
+    def broken(self):
+        clear_flag(self.expand_name('{endpoint_name}.available'))
+
+    def configure(self, mountpoint, fstype, options):
+        for relation in self.relations:
+            relation.to_publish_raw.update({
```

Comment:
This sends all mounts to all related apps, with most recent one wins.  Needs an `application` parameter and an `if` clause to only send the mount to the correct related app.